src/data_structures/interval_tree.rs

```diff
@@ -272,13 +255,24 @@ fn swap_interval_data<N, D>(node_1: &mut Node<N, D>, node_2: &mut Node<N, D>) {
 }
 
 fn intersect<N: Ord>(range_1: &Range<N>, range_2: &Range<N>) -> bool {
-    range_1.end > range_2.start && range_1.start < range_2.end
+    range_1.start < range_1.end && range_2.start < range_2.end &&
```

Comment:
That's true. So maybe let's keep it like it is for now, until the matter with RangeArgument has stabilized. There is also the intervallum crate, which might be usable with stable rust at some point.